trio/_threads.py

```diff
@@ -197,8 +198,10 @@ async def to_thread_run_sync(  # type: ignore[misc]
       sync_fn: An arbitrary synchronous callable.
       *args: Positional arguments to pass to sync_fn. If you need keyword
           arguments, use :func:`functools.partial`.
-      cancellable (bool): Whether to allow cancellation of this operation. See
-          discussion below.
+      abandon_on_cancel (bool): Whether to abandon this thread upon
+          cancellation of this operation. See discussion below.
+      cancellable (bool): *Deprecated* synonym for ``abandon_on_cancel``.
```

Comment:
Please remove discussion of `cancellable` from the documentation (deprecated options don't get docs), and raise a deprecation warning if it is passed. You will need to default it to None in order to notice if someone passed False explicitly. You can make it a hard error to supply both `cancellable` and `abandon_on_cancel`, so you don't have to worry about which one overrides the other one.